concent_api/core/validation.py

```diff
@@ -39,38 +32,34 @@ def validate_int_value(value):
         try:
             value = int(value)
         except (ValueError, TypeError):
-            raise Http400(
+            raise ConcentValidationError(
                 "Wrong type, expected a value that can be converted to an integer.",
                 error_code=ErrorCode.MESSAGE_VALUE_NOT_INTEGER,
             )
-    if value < 0:
-        raise Http400(
-            "Wrong type, expected non-negative integer but negative integer provided.",
-            error_code=ErrorCode.MESSAGE_VALUE_WRONG_TYPE,
-        )
+    validate_positive_integer_value(value)
 
 
 def validate_id_value(value, field_name):
     if not isinstance(value, str):
-        raise Http400(
+        raise ConcentValidationError(
```

Comment:
You can replace this with `validate_expected_value_type`.

Reply:
Done